rm/rm-server/src/main/java/org/ow2/proactive/resourcemanager/nodesource/NodeSource.java

```diff
@@ -743,10 +733,10 @@ public void finishNodeSourceShutdown(Client initiator) {
      * @return the list of deploying nodes handled by the infrastructure manager
      */
     @ImmediateService
-    public LinkedList<RMDeployingNode> getDeployingNodes() {
-        LinkedList<RMDeployingNode> result = new LinkedList<>();
-        result.addAll(this.infrastructureManager.getDeployingNodes());
-        return result;
+    public LinkedList<RMNodeData> getDeployingNodes() {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') The return type of this method should be an interface such as "List" rather than the implementation "LinkedList". [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1319)